x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/InternalTopMetrics.java

```diff
@@ -171,23 +171,23 @@ public boolean equals(Object obj) {
     }
 
     @Override
-    public final double sortValue(String key) {
+    public final SortValue sortValue(String key) {
         int index = metricNames.indexOf(key);
         if (index < 0) {
             throw new IllegalArgumentException("unknown metric [" + key + "]");
         }
         if (topMetrics.isEmpty()) {
-            return Double.NaN;
+            return SortValue.from(Double.NaN);
         }
 
         MetricValue value = topMetrics.get(0).metricValues.get(index);
         if (value == null) {
-            return Double.NaN;
+            return SortValue.from(Double.NaN);
         }
 
         // TODO it'd probably be nicer to have "compareTo" instead of assuming a double.
```

Comment:
You are doing this TODO I think. Sort of. You can't making a `comparteTo` for the bucket itself - but you are supporting non-double sort values.

Reply:
I was thinking of doing that instead of using Double.NaN.

Reply:
I don't really understand. When I wrote this TODO I thought it might be better to have the things implement `Comparable` but never did it. I think you've solved this is a simpler way.

Reply:
I wanted to use `Comparable` because in general I prefer to sue standard java stuff. Using `Comparable` would result in less coupling through a common dependency like `SortValue`. I will see if I can use something like `Comparable<?>` instead of `SortValue`.

Reply:
Yes, initially I wanted to use a `Comparable<Bucket>` but at that layer there is no information about the Bucket...so I need to use directly `SortValue`.